privacyidea/lib/crypto.py

```diff
@@ -307,86 +310,92 @@ def decryptPin(cryptPin):
 
 
 @log_with(log, log_entry=False)
-def encrypt(data, iv, id=0):
-    '''
+def encrypt(data, iv, key_id=0):
+    """
     encrypt a variable from the given input with an initialisation vector
 
-    :param data: buffer, which contains the value
-    :type  data: bytes or str
-    :param iv:   initialisation vector
-    :type  iv:   bytes or str
-    :param id:   contains the key id of the keyset which should be used
-    :type  id:   int
-    :return:     encrypted and hexlified data
+    :param data:   buffer, which contains the value
+    :type  data:   bytes or str
+    :param iv:     initialisation vector
+    :type  iv:     bytes or str
+    :param key_id: contains the key id of the keyset which should be used
+    :type  key_id: int
+    :return: encrypted and hexlified data
     :rtype: str
-
-    '''
+    """
     hsm = get_hsm()
-    ret = hsm.encrypt(to_bytes(data), to_bytes(iv), id)
+    ret = hsm.encrypt(to_bytes(data), to_bytes(iv), key_id)
     return hexlify_and_unicode(ret)
 
 
 @log_with(log, log_exit=False)
-def decrypt(input, iv, id=0):
-    '''
-    decrypt a variable from the given input with an initialiation vector
-
-    :param input: buffer, which contains the crypted value
-    :type  input: bytes or str
-    :param iv:    initialisation vector
-    :type  iv:    bytes or str
-    :param id:    contains the key id of the keyset which should be used
-    :type  id:    int
-    :return:      decrypted buffer
+def decrypt(enc_data, iv, key_id=0):
+    """
+    decrypt a variable from the given input with an initialisation vector
+
+    :param enc_data: buffer, which contains the encrypted value
+    :type  enc_data: bytes or str
+    :param iv:       initialisation vector
+    :type  iv:       bytes or str
+    :param key_id:   contains the key id of the keyset which should be used
+    :type  key_id:   int
+    :return: decrypted buffer
     :rtype: bytes
-    '''
+    """
     hsm = get_hsm()
-    res = hsm.decrypt(to_bytes(input), to_bytes(iv), id)
+    res = hsm.decrypt(to_bytes(enc_data), to_bytes(iv), key_id)
     return res
 
 
 @log_with(log, log_exit=False)
-def aes_decrypt(key, iv, cipherdata, mode=AES.MODE_CBC):
+def aes_decrypt(key, iv, cipherdata):
```

Comment:
Again, here the *data_input* or *enc_data* is called *cipherdata*. Time for clean up?

Is it worth to have this function utilize CBC and ECB?
Or is it not worth, since ECB is only used within the secretObj?
And ousite always CBC is used?

Reply:
on it.
This is why it is still a draft PR :-)
The `SecretObj` is used to store the cryptographic material of tokens. The material itself is encrypted/decrypted using the systems enckey or HSM.
Since the ECB-stuff is only used by the yubikey tokens, we should not mix it here.
And maybe we want to switch to a different encryption/decryption method for the cryptographic material (EC, something with MAC).